core/synch.c

```diff
@@ -1922,10 +1922,10 @@ detach_on_permanent_stack(bool internal, bool do_cleanup)
     DEBUG_DECLARE(bool ok;)
     DEBUG_DECLARE(int exit_res;)
     /* synch-all flags: if we fail to suspend a thread (e.g., privilege
-     * problems) ignore it.  XXX Should we retry instead?
+     * problems) retry it.
      */
     /* i#297: we only synch client threads after process exit event. */
-    uint flags = THREAD_SYNCH_SUSPEND_FAILURE_IGNORE | THREAD_SYNCH_SKIP_CLIENT_THREAD;
+    uint flags = THREAD_SYNCH_SUSPEND_FAILURE_RETRY | THREAD_SYNCH_SKIP_CLIENT_THREAD;
```

Comment:
I am afraid this is going to cause problems on Windows where it is not uncommon to have injected threads (CTRL_SHUTDOWN, CTRL_LOGOFF, etc.) we have no privileges to suspend -- and thus retrying will just fail again, and with the new "synchall failure is fatal and should kill the process" approach it turns what used to work for us on Windows into process death.

Reply:
How about exposing a flag option about whether or not this should be IGNORE, FATAL, or RETRY?

Reply:
OK, I think the main thing for me is that it seems safest to keep Windows w/ a default of ignore.

Reply:
OK, then I went with compile-time constants which keep windows functionality the same but updates Unix-based platforms.